### PR TITLE
Print table for benchmarks

### PR DESCRIPTION
**Pull request by jemmaissroff**
This PR changes the benchmarks to use the json blob that [benchmark-ips](https://github.com/evanphx/benchmark-ips) exposes to print the benchmarks at the end so we can copy / paste that table for the README

**Before merging:**

- [x] Copy the latest benchmark results into the `README.md` and update this PR
- [ ] If this change merits an update to `CHANGELOG.md`, add an entry following Keep a Changelog [guidelines](https://keepachangelog.com/en/1.0.0/) with [semantic versioning](https://semver.org/)

### CODE REVIEW

**Comment by PikachuEXE** [comment]
How do we make sure it's updated regularly to upstream `benchmark-ips` though?
Include version number in generated text to remind the benchmark runner?

**Comment by JacobEvelyn** [comment]
> How do we make sure it's updated regularly to upstream `benchmark-ips` though?

@jemmaissroff do you think we could get this feature merged into `benchmark-ips`?

**Comment by PikachuEXE** [comment]
I think it's ok to merge this before this feature got merged into official `benchmark-ips`.
But a way to remind updating the forked `benchmark-ips` is good to have.

**Review by JacobEvelyn — APPROVED** [review]
LGTM if you copy these benchmarks into the README and add back the sentence around what 1.00x means.

File: README.md
Comment (JacobEvelyn) [inline README.md:109]: I would vote to use language similar to what we had before, and cut this sentence and then say:

```
**Values >1.00x represent how much _slower_ each gem’s memoized value retrieval is than the latest commit of `MemoWise`.**
```

File: README.md
Comment (JacobEvelyn) [inline README.md:127]: We can cut this.

File: README.md
Comment (JacobEvelyn) [inline README.md:111]: Here's one potential way to condense this:

```suggestion
## Benchmarks

Benchmarks are run in GitHub Actions, and the tables below are updated with every code change. **Values >1.00x represent how much _slower_ each gem’s memoized value retrieval is than the latest commit of `MemoWise`**, according to [`benchmark-ips`](https://github.com/evanphx/benchmark-ips) (2.9.1).

Results using Ruby 3.0.2:

|Method arguments|`Memery` (1.4.0)|`Dry::Core` (0.7.1)|
|--|--|--|
|`()` (none)|1.14x|16.81x|
|`(a)`|2.02x|9.72x|
|`(a, b)`|1.07x|4.90x|
|`(a:)`|1.79x|18.74x|
|`(a:, b:)`|1.03x|9.94x|
|`(a, b:)`|1.03x|9.88x|
|`(a, *args)`|1.99x|4.63x|
|`(a:, **kwargs)`|1.54x|5.97x|
|`(a, *args, b:, **kwargs)`|0.99x|2.81x|

Results using Ruby 2.7.4 (because these gems raise errors in Ruby 3.x):

|Method arguments|`DDMemoize` (1.0.0)|`Memoizer` (1.0.3)|`Memoized` (1.0.2)|`Memoist` (0.16.2)|
|--|--|--|--|--|
|`()` (none)|26.75x|2.62x|1.20x|3.39x|
|`(a)`|19.58x|14.27x|10.51x|11.83x|
|`(a, b)`|4.14x|2.88x|2.36x|2.68x|
|`(a:)`|27.76x|23.54x|19.87x|21.36x|
|`(a:, b:)`|6.57x|5.57x|4.66x|5.06x|
|`(a, b:)`|6.57x|5.01x|4.38x|4.82x|
|`(a, *args)`|5.38x|4.02x|3.40x|3.48x|
|`(a:, **kwargs)`|3.95x|3.26x|2.92x|3.11x|
|`(a, *args, b:, **kwargs)`|2.77x|2.43x|2.23x|2.25x|
```

**Review by JacobEvelyn — APPROVED** [review]

**Pull request merged**
